src/Components/SmartComponents/Systems/Systems.js

```diff
@@ -50,16 +46,16 @@ const SystemsHook = ({ intl, location }) => {
     const metadata = useSelector(
         ({ SystemsStore }) => SystemsStore.metadata
     );
+
+    const [selectedHosts, setSelectedHosts] = useState(undefined);
+
     const isLoading =  useSelector(({ SystemsStore }) => SystemsStore.isLoading);
 
     const items = useMemo(() => selectHosts(systems), [systems]);
 
-    const params = () => ({ filter,  page_size: pageSize, opt_out: optOut, page });
-
     useEffect(() => {
-        if (Object.keys(parameters).length !== 0 && parameters.constructor === Object) {
-            dispatch(fetchSystems(parameters));
-        }
+        !parameters.isFirstLoad && dispatch(fetchSystems(parameters));
+        parameters.urlParamsAllowed && createUrlParams(parameters);
```

Comment:
The same question, why do we need the extra flag?
`!parameters.isFirstLoad` wouldn't do the same?

Reply:
I am not able to implement my logic with one flag unfortunately.

Reply:
I have created a new logic based on how Patchman is handling the filtering and urlParams. This logic is more self-explanatory, I think. Previously, I did not want to use extra handleRefresh for inventory, thus I was using 2 flags in redux and mergeUrlParams function. It was not so clear what my logic was :)) . In this, logic also I am using two flags in react component state. First flag is used to prevent useEffect hook to fire unwanted dsipatch, next is used to prevent pushing hitsory new urlPrams. I have done quite a lot of testing. It believe workes as expected.